rating/peersRatingMonitor.go

```diff
@@ -41,48 +46,68 @@ func checkMonitorArgs(args ArgPeersRatingMonitor) error {
 	if check.IfNil(args.BadRatedCache) {
 		return fmt.Errorf("%w for BadRatedCache", p2p.ErrNilCacher)
 	}
+	if check.IfNil(args.ConnectionsProvider) {
+		return p2p.ErrNilConnectionsProvider
+	}
 
 	return nil
 }
 
-// GetPeersRatings returns the current ratings
-func (monitor *peersRatingMonitor) GetPeersRatings() string {
+// GetConnectedPeersRatings returns the ratings of the current connected peers
+func (monitor *peersRatingMonitor) GetConnectedPeersRatings() string {
 	ratingsMap := getRatings(monitor.topRatedCache)
```

Comment:
I think the overall function performance can be improved by first getting the connected peers list (which should be ~75 elements) and then taking the ratings out of the 2 caches.

Reply:
good suggestion, pushed